File: GMS-backend/GMS/Entities/Endpoint/Suplement/Edit/SuplementiEditEndpoint.cs
using GMS.Data;
using GMS.Helpers;
using GMS.Helpers.Auth;
using Microsoft.AspNetCore.Mvc;

namespace GMS.Entities.Endpoint.Suplement.Edit
{
    [Route("Suplement-Edit")]
    [MyAuthorization]

    public class SuplementiEditEndpoint : MyBaseEndpoint<SuplementiEditRequest, int>
    {
        private readonly ApplicationDbContext db;

        public SuplementiEditEndpoint(ApplicationDbContext db)
        {
            this.db = db;
        }

        [HttpPost]
        public override async Task<int> Handle([FromBody]SuplementiEditRequest request, CancellationToken cancellationToken)
        {
            Models.Suplement? suplement;
            if (request.ID == 0)
            {
                suplement = new Models.Suplement();
                db.Add(suplement);


            }
            else
            {
                suplement = db.Suplement.FirstOrDefault(s => s.ID == request.ID);
                if (suplement == null)
                    throw new Exception("pogresan ID");
            }

            suplement.Naziv = request.Naziv.RemoveTags();
            suplement.Cijena = request.Cijena;
            suplement.Gramaza = request.Gramaza;
            suplement.Opis = request.Opis.RemoveTags();
            suplement.Slika = request.Slika.RemoveTags();


            await db.SaveChangesAsync(cancellationToken);

            return suplement.ID;
            
        }
    }
}

File: GMS-frontend/src/app/endpoints/suplementi-endpoints/suplementi-edit-endpoint.ts

import {Injectable} from "@angular/core";
import {HttpClient} from "@angular/common/http";
import {MyBaseEndpoint} from "../MyBaseEndpoint";
import {Config} from "../../config";
import {Observable} from "rxjs";
@Injectable({providedIn: 'root'})
export class SuplementiEditEndpoint implements  MyBaseEndpoint<SuplementiEditRequest, number>{
  constructor(public httpClient:HttpClient) { }

  Handle(request: SuplementiEditRequest): Observable<number> {
    let url=Config.adresa + `Suplement-Edit`;

    return this.httpClient.post<number>(url, request);
  }

}
export interface SuplementiEditRequest {
  id: number
  naziv: string
  cijena: number
  gramaza: number
  opis: string
  slika: string | undefined
  dobavljacID: number
  kategorijaID: number
}

